### PR TITLE
feat: add /thermostats/update endpoint

### PR DESCRIPTION
**Pull request by mikewuu**
live endpoint: https://github.com/seamapi/seam-connect/blob/main/pages/api/public/thermostats/update.ts

### CODE REVIEW

**Review by xplato — APPROVED** [review]

File: src/pages/api/thermostats/update.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      message: `Could not find a thermostat with device_id`,
+      data: { device_id },
+    })
+  }
+
+  const thermostat_original_climate_setting =
+    "default_climate_setting" in device.properties
+      ? device.properties.default_climate_setting
+      : null
+
```
Comment (Andrewjeska) [inline src/pages/api/thermostats/update.ts:45]: Should we add this check here?
```
if (
    !thermostat_original_climate_setting &&
    !default_climate_setting["manual_override_allowed"]
  )
    throw new Error("manual_override_allowed must be defined")
```

We check for this here: https://github.com/seamapi/seam-connect/blob/main/pages/api/public/thermostats/update.ts#L37C1-L42C1

File: src/lib/zod/device.ts [hunk 2]
```diff
@@ -66,6 +66,7 @@ export const lock_device_properties = common_device_properties.extend({
 })
 
 export const thermostat_device_properties = common_device_properties.extend({
+  manufacturer: z.string().optional(),
```
Comment (Andrewjeska) [inline src/lib/zod/device.ts:69]: This should be a common device property, see here: 

https://github.com/seamapi/seam-connect/blob/main/public-types/unstable/models/managed-device.ts#L28

File: src/pages/api/thermostats/update.ts (hunk 1 above)
Comment (mikewuu) [reply src/pages/api/thermostats/update.ts:45]: thanks, updated in https://github.com/seamapi/fake-seam-connect/pull/101/commits/b90b964990302bb866aaeb9b2a4f23128c3a8b1c

File: src/lib/zod/device.ts (hunk 2 above)
Comment (mikewuu) [reply src/lib/zod/device.ts]: updated in https://github.com/seamapi/fake-seam-connect/pull/101/commits/cd2e39d2e43a7cfdd787f63f03b3584e5e1748a4

**Pull request merged**